network/dag/transaction.go

```diff
@@ -74,6 +74,7 @@ type NetworkHeader interface {
 
 // SubscriberTransaction contains a subset of the Transaction methods which are only relevant to subscribers.
 type SubscriberTransaction interface {
+	NetworkHeader
```

Comment:
If we keep the 'subscriber' way of calculating blocks, it needs the `NetworkHeader`, which defeats the purpose of having a `SubscriberTransaction` (separate from `Transaction`). So this will have to go in that case.